recipes/libiconv/all/conanfile.py

```diff
@@ -55,99 +63,66 @@ def configure(self):
-            lib = "lib" if is_msvc(self) else os.environ.get("AR", "llvm-lib")
-            build_aux_path = os.path.join(self.build_folder, self._source_subfolder, "build-aux")
-            lt_compile = tools.unix_path(os.path.join(build_aux_path, "compile"))
-            lt_ar = tools.unix_path(os.path.join(build_aux_path, "ar-lib"))
-            env_vars.update({
-                "CC": "{} {} -nologo".format(lt_compile, cc),
-                "CXX": "{} {} -nologo".format(lt_compile, cxx),
-                "LD": "link",
-                "STRIP": ":",
-                "AR": "{} {}".format(lt_ar, lib),
-                "RANLIB": ":",
-                "NM": "dumpbin -symbols"
-            })
-            env_vars["win32_target"] = "_WIN32_WINNT_VISTA"
-
-        if not tools.cross_building(self) or is_msvc(self) or self._is_clang_cl:
-            rc = None
-            if self.settings.arch == "x86":
-                rc = "windres --target=pe-i386"
-            elif self.settings.arch == "x86_64":
-                rc = "windres --target=pe-x86-64"
-            if rc:
-                env_vars["RC"] = rc
-                env_vars["WINDRES"] = rc
-        if self._use_winbash:
-            env_vars["RANLIB"] = ":"
-
-        with tools.vcvars(self.settings) if (is_msvc(self) or self._is_clang_cl) else tools.no_op():
-            with tools.chdir(self._source_subfolder):
-                with tools.environment_append(env_vars):
-                    yield
-
-    @functools.lru_cache(1)
-    def _configure_autotools(self):
-        host = None
-        build = None
-        if is_msvc(self) or self._is_clang_cl:
-            build = False
-            if self.settings.arch == "x86":
-                host = "i686-w64-mingw32"
-            elif self.settings.arch == "x86_64":
-                host = "x86_64-w64-mingw32"
-
-        autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-
-        configure_args = []
-        if self.options.shared:
-            configure_args.extend(["--disable-static", "--enable-shared"])
-        else:
-            configure_args.extend(["--enable-static", "--disable-shared"])
+            cc, lib, link = self._msvc_tools
+            build_aux_path = os.path.join(self.source_folder, "build-aux")
+            lt_compile = unix_path(self, os.path.join(build_aux_path, "compile"))
+            lt_ar = unix_path(self, os.path.join(build_aux_path, "ar-lib"))
+            env.define("CC", f"{lt_compile} {cc} -nologo")
+            env.define("CXX", f"{lt_compile} {cc} -nologo")
+            env.define("LD", f"{link}")
+            env.define("STRIP", ":")
+            env.define("AR", f"{lt_ar} {lib}")
+            env.define("RANLIB", ":")
+            env.define("NM", "dumpbin -symbols")
+            env.define("win32_target", "_WIN32_WINNT_VISTA")
+
+        if is_msvc(self) and Version(self.settings.compiler.version) >= "12":
```

Comment:
This isn't going to work correctly for the `msvc` compiler, only `Visual Studio`. Unfortunately, there won't be a more helpful solution until Conan 1.53.0 for this: conan-io/conan/#11158

Reply:
Good catch!